modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -63,73 +120,211 @@ def _unstrip_protocol(protocol, path):
+                with self.filesystem.open(file) as f:
+                    row_groups = ParquetFile(f).num_row_groups
+                    row_groups_per_file.append(row_groups)
+            self._row_groups_per_file = row_groups_per_file
+        return self._row_groups_per_file
+
+    @property
+    def files(self):
+        if self._files is None:
+            self._files = self._get_files(self.dataset.files)
+        return self._files
+
+    def read_table_as_pandas(
+        self,
+        columns,
+    ):
+        from pyarrow.parquet import read_table
+
+        return read_table(
+            self.path, columns=columns, filesystem=self.filesystem
+        ).to_pandas()
+
+
+@_inherit_docstrings(Dataset)
+class FastParquetDataset(Dataset):
+    def __init__(self, path, storage_options):
+        self.path = path
+        self.storage_options = storage_options
+        self.dataset = self._init_dataset()
+        self._row_groups_per_file = None
+        self.filesystem, self._files = self._get_files(self._get_fastparquet_files())
+
+    def _init_dataset(self):  # noqa: GL08
+        from fastparquet import ParquetFile
+
+        fs, fs_path = self._get_fs_and_fs_path()
+        dataset = ParquetFile(fs_path, fs=fs)
+        return dataset
+
+    @property
+    def pandas_metadata(self):
+        import json
+
+        if "pandas" not in self.dataset.key_value_metadata:
+            return {}
+        return json.loads(self.dataset.key_value_metadata["pandas"])
+
+    @property
+    def columns(self):
+        return self.dataset.columns
+
+    @property
+    def engine(self):
+        return "fastparquet"
+
+    def row_groups_per_file(self):
+        from fastparquet import ParquetFile
+
+        if self._row_groups_per_file is None:
+            row_groups_per_file = []
+            # Count up the total number of row groups across all files and
+            # keep track of row groups per file to use later.
+            for file in self.files:
+                with self.filesystem.open(file) as f:
+                    row_groups = ParquetFile(f).info["row_groups"]
+                    row_groups_per_file.append(row_groups)
+            self._row_groups_per_file = row_groups_per_file
+        return self._row_groups_per_file
+
+    @property
+    def files(self):
+        return self._files
+
+    def read_table_as_pandas(self, columns):
+        return self.dataset.to_pandas(columns=columns)
+
+    def _get_fastparquet_files(self):  # noqa: GL08
+        # fastparquet doesn't have a nice method like PyArrow, so we
+        # have to copy some of their logic here while we work on getting
+        # an easier method to get a list of valid files.
+        # See: https://github.com/dask/fastparquet/issues/795
+        if isinstance(self.path, AbstractBufferedFile):
+            fs = self.path.fs
+        else:
+            fs = get_filesystem_class(split_protocol(self.path)[0])(
+                **self.storage_options
+            )
```

Comment:
@mvashishtha ditto comment from before

Reply:
I see. Let's discuss in the other thread.